File: src/chunk.go
// chunk.go
//
// Copyright (c) 2026 PDFjet Software
// Licensed under the MIT License. See LICENSE file in the project root.

package pdfjet

// pngChunk is used by the pngimage module.
type pngChunk struct {
	chunkLength uint32
	chunkType   []byte
	chunkData   []byte
	chunkCRC    uint32
}

// newPNGChunk construct new PNG chunk.
func newPNGChunk() *pngChunk {
	chunk := new(pngChunk)
	return chunk
}

File: Sources/PDFjet/Chunk.swift
/**
 * Chunk.swift
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
import Foundation

class Chunk {
    var length: UInt32?
    var type: [UInt8]?
    var data: [UInt8]?
    var crc: UInt32?

    func getData() -> [UInt8]? {
        return self.data
    }

    func setData(_ data: [UInt8]?) {
        self.data = data
    }

}   // End of Chunk.swift
